telluric/georaster.py

```diff
@@ -614,7 +614,7 @@ def __eq__(self, other):
             and self.shape == other.shape \
             and self.image.dtype == other.image.dtype \
             and np.array_equal(self.image.mask, other.image.mask) \
-            and np.array_equal(self.image, other.image)
+            and np.array_equal(np.ma.filled(self.image, 0), np.ma.filled(other.image, 0))
```

Comment:
why this change?

Reply:
The reason is that now we are treating the nodata value as an implementation detail, and `np.array_equal` compares both `image.data` and `image.mask`. What I used for the comparison instead is `np.ma.filled`, which fills all the nodata values in the same way:

```
In [1]: import numpy as np

In [2]: arr1 = np.ma.masked_array([0, 0, 1, 2, 3], [True, False, False, False, False])

In [3]: arr2 = np.ma.masked_array([99, 0, 1, 2, 3], [True, False, False, False, False])

In [4]: np.array_equal(arr1, arr2)
Out[4]: False

In [5]: np.ma.filled(arr1, 0)
Out[5]: array([0, 0, 1, 2, 3])

In [6]: np.ma.filled(arr2, 0)
Out[6]: array([0, 0, 1, 2, 3])

In [7]: np.array_equal(np.ma.filled(arr1, 0), np.ma.filled(arr2, 0))
Out[7]: True
```

Reply:
nice!!
so, this is the way to extract the information of a masked array

Reply:
might this be expensive performance wise when comparing large images ?

Reply:
With a 10000 x 10000 masked array (quite big), it takes 1 second:

```
In [13]: arr2 = np.ma.masked_array(np.random.randn(10000, 10000), np.random.randint(2, size=(10000, 10000)).astype(bool))

In [14]: %timeit np.ma.filled(arr2, 0)
1.2 s ± 8.33 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

don't think it's going to be a bottleneck, creating the array itself took more time:

```
In [17]: %timeit np.ma.masked_array(np.random.randn(10000, 10000), np.random.randint(2, size=(10000, 10000)).astype(bool))
6.62 s ± 172 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```